lib/fit/file/data.rb

```diff
@@ -21,14 +21,18 @@ def record_type
 
           definition.fields.each do |field|
             class_eval <<-RUBY, __FILE__, __LINE__ + 1
-              # string are not null terminated when they have the field length
-              #{field.type} :#{field.raw_name} #{ ", :read_length => #{field.size}, :trim_padding => true" if field.type == "string" }
-
-              # some cases found where a field has several time the normal size (multiple elements ?)
-              #{ if(field.type != "string" and field.size > field.length) then n=field.size/field.length; (2..n).inject("") { |acc, i| acc + "\n#{field.type} :#{field.raw_name}__#{i.to_s}" } end }
+              # in case the field size is a multiple of the field lenght, we must build an array
+              #{
+              if (field.type != "string" and field.size > field.length)
+                "array :#{field.raw_name}, :type => :#{field.type}, :initial_length => #{field.size/field.length}"
+              else
+                # string are not null terminated when they have the field length
+                "#{field.type} :#{field.raw_name} #{ ", :read_length => #{field.size}, :trim_padding => true" if field.type == "string" }"
+              end
+              }
```

Comment:
I think it would be cleaner to move the conditions outside of `class_eval`, what do you think?

```ruby
code = ""

if field.type != "string" && field.size > field.length
  # in case the field size is a multiple of the field length, we must build an array
  code << "array :#{field.raw_name}, :type => :#{field.type}, :initial_length => #{field.size / field.length}"
else
  # string are not null terminated when they have the field length
  code << "#{field.type} :#{field.raw_name} #{ ", :read_length => #{field.size}, :trim_padding => true" if field.type == "string" }"
end

code << <<-RUBY
  def #{field.name}
    #{field.raw_name}.snapshot #{ "/ #{field.scale.inspect}.0" if field.scale }
  end
RUBY

class_eval code, __FILE__, __LINE__ + 1
```